src/System.Device.Gpio/System/Device/ComponentInformation.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+#pragma warning disable CS1591
+namespace System.Device
+{
+    public record ComponentInformation
+    {
+        private readonly List<ComponentInformation> _subComponents;
+
+        public ComponentInformation(object instance, string name)
+            : this(instance, name, string.Empty)
+        {
+        }
+
+        public ComponentInformation(object instance, string name, string version)
+            : this(instance, name, version, ComponentState.Unknown)
+        {
+        }
+
+        public ComponentInformation(object instance, string name, string version, ComponentState componentState)
+        {
```

Comment:
What about using `Version` type instead of a string? Does it fit with this scenario?
My point is that would be much easier to compare

Reply:
I considered this, unfortunately the `Version` struct does not support modern semantic Versioning with Version tags.